stock_picking_auto_revert/models/stock_move.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2019 ForgeFlow
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import _, exceptions, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _check_restrictions(self):
+        # Restrictions before remove quants
+        if self.returned_move_ids:
+            raise exceptions.UserError(
+                _("Action not allowed. Move splited / with returned moves.")
+            )
+        if self.move_dest_ids or self.search([("move_dest_ids", "in", self.ids)]):
+            raise exceptions.UserError(_("Action not allowed. Chained move."))
```

Comment:
A better user message could be great.

`You cannot revert this stock picking. Its stock moves are linked to other moves. Origin pickings: %(origin_pickings)s. Destination pickings: %(destination_pickings)%`

Reply:
Much better! thanks